FirebaseRemoteConfig/SwiftNew/ConfigContent.swift

```diff
@@ -12,8 +12,510 @@
+/// config or persisted in database.
+@objc(RCNConfigContent) public
+class ConfigContent: NSObject {
+  /// Active config data that is currently used.
+  private var _activeConfig: [String: [String: RemoteConfigValue]] = [:]
+
+  /// Pending config (aka Fetched config) data that is latest data from server that might or might
+  /// not be applied.
+  private var _fetchedConfig: [String: [String: RemoteConfigValue]] = [:]
+
+  /// Default config provided by user.
+  private var _defaultConfig: [String: [String: RemoteConfigValue]] = [:]
+
+  /// Active Personalization metadata that is currently used.
+  private var _activePersonalization: [String: Any] = [:]
+
+  /// Pending Personalization metadata that is latest data from server that might or might not be
+  /// applied.
+  private var _fetchedPersonalization: [String: Any] = [:]
+
+  /// Active Rollout metadata that is currently used.
+  private var _activeRolloutMetadata: [[String: Any]] = []
+
+  /// Pending Rollout metadata that is latest data from server that might or might not be applied.
+  private var _fetchedRolloutMetadata: [[String: Any]] = []
+
+  /// DBManager
+  private var dbManager: ConfigDBManager?
+
+  /// Current bundle identifier;
+  private var bundleIdentifier: String
+
+  /// Blocks all config reads until we have read from the database. This only
+  /// potentially blocks on the first read. Should be a no-wait for all subsequent reads once we
+  /// have data read into memory from the database.
+  private let dispatchGroup: DispatchGroup
+
+  /// Boolean indicating if initial DB load of fetched,active and default config has succeeded.
+  private var isConfigLoadFromDBCompleted: Bool
+
+  /// Boolean indicating that the load from database has initiated at least once.
+  private var isDatabaseLoadAlreadyInitiated: Bool
+
+  /// Default timeout when waiting to read data from database.
+  private let databaseLoadTimeoutSecs = 30.0
+
+  /// Shared Singleton Instance
+  @objc public
+  static let sharedInstance = ConfigContent(dbManager: ConfigDBManager.sharedInstance)
+
+  /// Designated initializer
+  @objc(initWithDBManager:) public
+  init(dbManager: ConfigDBManager) {
+    self.dbManager = dbManager
+    bundleIdentifier = Bundle.main.bundleIdentifier ?? ""
+    if bundleIdentifier.isEmpty {
+      RCLog.notice("I-RCN000038",
+                   "Main bundle identifier is missing. Remote Config might not work properly.")
+    }
+    dispatchGroup = DispatchGroup()
+    isConfigLoadFromDBCompleted = false
+    isDatabaseLoadAlreadyInitiated = false
+    super.init()
+    loadConfigFromMainTable()
+  }
+
+  // Blocking call that returns true/false once database load completes / times out.
+  // @return Initialization status.
+  @objc public
+  func initializationSuccessful() -> Bool {
+    assert(!Thread.isMainThread, "Must not be executing on the main thread.")
+    return checkAndWaitForInitialDatabaseLoad()
+  }
+
+  @discardableResult
```

Comment:
grabbing missing comment from objc impl.
```suggestion
  /// We load the database async at init time. Block all further calls to active/fetched/default
  /// configs until load is done.
  @discardableResult
```